main/com.tunein/device.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const Tp = require('thingpedia');
+const axios = require('axios');
+const axiosExtensions = require('axios-extensions');
+const url = require('url');
+
+const BASE_URL = 'http://opml.radiotime.com';
+const CONTENT_TYPE = 'application/x-www-form-urlencoded';
+const HTTP_STATUS_OK = 200;
+const SEARCH_PARAM = '/Search.ashx';
+const BROWSE_PARAM = '/Browse.ashx';
+const BROWSE_ITEMS = ['trending', 'music', 'sports', 'talk'];
+
+module.exports = class TuneinRadioDevice extends Tp.BaseDevice {
+    constructor(engine, state, args={}) {
+         super(engine, state);
+         this.uniqueId = 'com.tunein';
+         this.name = 'Tunein';
+         this.description = 'Stream radio through TuneIn API';
+         axios.defaults.headers.post['Content-Type'] = args.content_type || CONTENT_TYPE;
+         axios.defaults.baseURL = args.protocol || BASE_URL;
+         axios.defaults.params = { render: 'json' };
+    }
+
+    _get_tunein_response(req) {
+        return new Promise( (resolve, reject) => {
+            axios.get(req.url, req)
+                .then(response => { 
```

Comment:
If you're working with a promise based API, you don't need to use the `Promise` constructor, and you don't need this whole convoluted code, just write an `async` function.